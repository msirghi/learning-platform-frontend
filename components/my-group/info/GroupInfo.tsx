import React from 'react';
import styles from '../../../styles/modules/MyGroup.module.scss';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
import { IconButton } from '@material-ui/core';
import { useTranslation } from '../../../i18n';

type Props = {
  totalMembers: number;
  totalOnline: number;
  groupName: string;
  showChevron?: boolean;
  chevronDown?: boolean;
  onChevronClick?: () => void;
};

export const GroupInfo: React.FC<Props> = ({
  groupName,
  totalMembers,
  totalOnline,
  showChevron,
  chevronDown,
  onChevronClick
}) => {
  const { t } = useTranslation();

  return (
    <div className={styles.groupInfoContainer}>
      <div>
        <div className={styles.header}>{groupName}</div>
        <div className={styles.subheader}>
          {totalMembers} {t('myGroup:members')} / {totalOnline} {t('myGroup:online')}
        </div>
      </div>
      {showChevron && (
        <div className={styles.chevron}>
          <IconButton onClick={() => onChevronClick()}>
            {chevronDown ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          </IconButton>
        </div>
      )}
    </div>
  );
};
