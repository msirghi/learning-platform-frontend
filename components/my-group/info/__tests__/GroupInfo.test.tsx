import { IconButton } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { GroupInfo } from '../GroupInfo';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';

describe('GroupInfo component', () => {
  const props = {
    totalMembers: 1,
    totalOnline: 1,
    groupName: 'name',
    showChevron: true,
    chevronDown: true,
    onChevronClick: jest.fn()
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<GroupInfo {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire on click prop handler on icon button click', () => {
    const wrapper = shallow(<GroupInfo {...props} />);
    const iconButton = wrapper.find(IconButton);
    expect(iconButton).toHaveLength(1);

    iconButton.simulate('click');
    expect(props.onChevronClick).toBeCalled();
  });

  it('should render arrow up on chevron down status', () => {
    const wrapper = shallow(<GroupInfo {...props} />);
    const arrowUp = wrapper.find(KeyboardArrowUpIcon);
    const arrowDown = wrapper.find(KeyboardArrowDownIcon);

    expect(arrowUp).toHaveLength(1);
    expect(arrowDown).toHaveLength(0);
  });

  it('should render arrow down on chevron up status', () => {
    const wrapper = shallow(<GroupInfo {...props} chevronDown={false} />);
    const arrowUp = wrapper.find(KeyboardArrowUpIcon);
    const arrowDown = wrapper.find(KeyboardArrowDownIcon);

    expect(arrowUp).toHaveLength(0);
    expect(arrowDown).toHaveLength(1);
  });
});
