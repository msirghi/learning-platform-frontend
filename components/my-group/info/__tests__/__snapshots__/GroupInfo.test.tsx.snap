// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupInfo component should match the snapshot 1`] = `
<div
  className="groupInfoContainer"
>
  <div>
    <div
      className="header"
    >
      name
    </div>
    <div
      className="subheader"
    >
      1
       
      myGroup:members
       / 
      1
       
      myGroup:online
    </div>
  </div>
  <div
    className="chevron"
  >
    <WithStyles(ForwardRef(IconButton))
      onClick={[Function]}
    >
      <Memo([object Object]) />
    </WithStyles(ForwardRef(IconButton))>
  </div>
</div>
`;
